### PR TITLE
CSS3_Box_Sizing_Models

### LINKED ISSUE #1221: Box Sizing Models CSS
<hr>

## Description 📜

I want to create documentation on box sizing models HTML/CSS

<hr>

## Domain of Contribution 📊

- [x] Frontend Dev (HTML, CSS, JS)

<hr>

### PR DESCRIPTION
**Pull request by onlykingKD**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
I have tried to explain Box Sizing in CSS in a detailed form with images and code included.
Fixes #1221 

<hr>

## Type of change 📝
Added a detailed documentation of Box Sizing Models CSS including the code snippets.

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Frontend Dev HTML/CSS/JS
<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
Add reference of each image. Rest everything looks good. @praveenscience

**Comment by onlykingKD** [comment]
Reference Added !

**Comment by kashishahuja2002** [comment]
> Reference Added !

Perfect.

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
Add references as link, not text.

**Comment by onlykingKD** [comment]
Added 👍🏻

**Review by kashishahuja2002 — COMMENTED** [review]
Good to go. @praveenscience

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Box_Sizing_CSS
+This is my first contribution as Open Source (thanks to GWOC) and here I will be expalining in detail about the various CSS Box Sizing Types.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:2]: Need space after heading.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+}`
+
+Vendor-prefixes for box-sizing:  
+`*{ 
+  -webkit-box-sizing: border-box; 
+  -moz-box-sizing: border-box;
+  box-sizing: border-box; 
+}`
+
+**Reference**
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:111]: Make this a heading.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
+# Box_Sizing_CSS
+This is my first contribution as Open Source (thanks to GWOC) and here I will be expalining in detail about the various CSS Box Sizing Types.  
+
+Before we proceed any further first of all we need to know what a box model is.
+
+## What is a box model? 
+
+We can imagine a box model as a rectangular box that wraps around each HTML element. It basically comprises of the padding, margin border and content inside the element.
+We all know that all elements on a webpage is represnted as rectangular box.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:9]: Spaces after paragraph.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
+# Box_Sizing_CSS
+This is my first contribution as Open Source (thanks to GWOC) and here I will be expalining in detail about the various CSS Box Sizing Types.  
+
+Before we proceed any further first of all we need to know what a box model is.
+
+## What is a box model? 
+
+We can imagine a box model as a rectangular box that wraps around each HTML element. It basically comprises of the padding, margin border and content inside the element.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:8]: Spaces after paragraph.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+Let’s try understand what each of these properties signifies:
+
+**Content:** This is the main part of the box where the content put by the user appears. 
+
+**Padding:** This space is around the content inside an element (transparent).
+
+**Border:** This  is self expalnatory. It usually goes around the padding and content.
+
+**Margin:** This the space around the box (also tranparent).
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:25]: Can we make this into an unordered list?

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+Thus, so as to urge the specified width and height as laid out in the planning for the element that's rendered on the DOM, we frequently recalculate the width and height by subtracting the padding and border, either manually or by using the calc(width-valuex) function, (where valuex is that the sum of the padding and/or border assigned thereto element) from their respective dimensions.
+
+Here the actual width of the element is 300px, but since we assign border and padding to it, we manually recalculate the width as
+
+`my-div{
+  width: 350px;
+  border: solid 5px #82B366;
+  padding: 20px;
+}`
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:58]: Make it a code fence.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+width = 300–40(padding-right and padding-left)–10(border-right and border-left) or we can also use the CSSS calc function.
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:60]: Put this inside inline code.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+This is not a good practice as it doesn’t make it very obvious when we see that the both the width differ : assigned width is 250px, while the rendered width is 300px. Here it confuses the programmer and it arouses the need for applying the box-sizing property to an element.
+
+We could also have increased values(width and height) when an element is rendered on the DOM if we do not recalculate the width and height of the element as in the above example. The demonstartion that we see below is that the total width rendered on the DOM is 350px( here the recalculation of the width isn’t need to be done)
+
+`my-div{
+  width: 300px;
+  border: solid 5px #82B366;
+  padding: 20px;
+}`
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:70]: Make it a code fence with syntax highlighting.
Indent the code properly.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+In these kinds of situations where we can make use of the CSS3 beautiful property “BOX-SIZING”.
+
+### What is box-sizing?
+
+Basically the box-sizing property tells the browser what the sizing properties (width and height) should include or not include.
+
+The CSS3 Box-sizing property has the following syntax:   
+
+> Box-sizing: content-box|border-box|padding-box|initial|inherit;
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:80]: Use inline code-blocks for the syntax.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+By default it takes the value content-box.  
+
+When we give **box-sizing: border box**, the special combination property called property-value is used in an element, which tells the browser to display the width property as the actual rendered width. Modifying the previous code as follows:
+
+`my-div{
+  box-sizing: border-box;
+  width: 300px;
+  border: solid 5px #82B366;
+  padding: 20px;
+}`
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:91]: Use Code fences.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+Basically the box-sizing property tells the browser what the sizing properties (width and height) should include or not include.
+
+The CSS3 Box-sizing property has the following syntax:   
+
+> Box-sizing: content-box|border-box|padding-box|initial|inherit;
+
+By default it takes the value content-box.  
+
+When we give **box-sizing: border box**, the special combination property called property-value is used in an element, which tells the browser to display the width property as the actual rendered width. Modifying the previous code as follows:
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:84]: Use inline code blocks.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+![image](https://user-images.githubusercontent.com/55577276/134493069-39a3ba69-c2d5-41cf-8a31-3e973e6f3490.png)
+
+*Figure 3: Difference in the rendered width by using box-sizing property (Reference : https://medium.com/prod-io)*
+
+We can also use the universal specifier specifing  box sizing as border box that will apply this to all the elements present in DOM.
+Universal box-sizing:  
+
+`*, *:before, *:after{  
+  box-sizing: border-box;  
+}`
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:102]: Use Code fences.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+Vendor-prefixes for box-sizing:  
+`*{ 
+  -webkit-box-sizing: border-box; 
+  -moz-box-sizing: border-box;
+  box-sizing: border-box; 
+}`
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:109]: Add space before code fence.
Use Code fences.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+**Reference**  
+
+https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing
+
+*Thank you !*
```
Comment (praveenscience) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:115]: No need of thank you!

**Review by praveenscience — CHANGES_REQUESTED** [review]
Please check.

**Comment by onlykingKD** [comment]
@praveenscience sir please review , all the changes you suggested I have completed them.

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+```css
+my-div{
+  width: 350px;
+  border: solid 5px #82B366;
+  padding: 20px;
+}
+```
+
+width = 300–40(padding-right and padding-left)–10(border-right and border-left) or we can also use the CSSS calc function.
```
Comment (ipsitagoel) [inline Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md:82]: @onlykingKD Add this inside a code block/as an inline code.

**Review by ipsitagoel — CHANGES_REQUESTED** [review]

**Review by kashishahuja2002 — APPROVED** [review]

**Comment by kashishahuja2002** [comment]
Great work @onlykingKD

File: Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md (hunk 15 above)
Comment (onlykingKD) [reply Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md]: donee

**Review by ipsitagoel — APPROVED** [review]
@onlykingKD Good job!
@2024-SANDHYA Have a look, it's good to merge.

**Review by 2024-SANDHYA — APPROVED** [review]

**Comment by onlykingKD** [comment]
@2024-SANDHYA maam can you please **hacktoberfest** label to my PR

**Pull request merged**